source/include/ota_platform_interface.h

```diff
@@ -192,6 +192,10 @@ typedef OtaPalStatus_t ( * OtaPalCloseFile_t )( OtaFileContext_t * const pFileCo
 /**
  * @brief Write a block of data to the specified file at the given offset.
  *
+ * Note: This function is supposed to write a complete file block before returning. So the number of
+ * bytes written will be equal to the size of the file block. Exception only in case of last block
+ * where the number of bytes written will be equal or less than the size of the file block.
+ *
```

Comment:
Shouldn't the caller verify this condition with an assertion?

Reply:
Hi Paul,
Updated at #435. Trigger assertion at this condition.

Thanks.